hazelcast/src/main/java/com/hazelcast/sql/impl/type/converter/AbstractDecimalConverter.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.type.converter;
+
+import com.hazelcast.sql.impl.type.QueryDataTypeFamily;
+
+public abstract class AbstractDecimalConverter extends Converter {
+    protected AbstractDecimalConverter(int id) {
+        super(id, QueryDataTypeFamily.DECIMAL);
+    }
+
+    @Override
+    public final Object asObject(Object val) {
+        return asDecimal(val);
```

Comment:
Why some converters are trying to convert the passed value to their own type while others are not doing that? For instance, `IntegerConverter` and many others are not trying to convert.

Reply:
Because there are several descendants of `AbstractDecimalConverter`, so we do not know the value of the passed `val` argument. Specifically, here it could be either `BitDecimal` or `BigInteger`.